src/integrations/settings-integration.php

```diff
@@ -258,10 +258,18 @@ public function register_hooks() {
 
 		// Are we saving the settings?
 		if ( $this->current_page_helper->get_current_admin_page() === 'options.php' ) {
-			// phpcs:disable WordPress.PHP.NoSilencedErrors.Discouraged -- This deprecation will be addressed later.
-			$post_action = \filter_input( \INPUT_POST, 'action', @\FILTER_SANITIZE_STRING );
-			$option_page = \filter_input( \INPUT_POST, 'option_page', @\FILTER_SANITIZE_STRING );
-			// phpcs:enable
+			$post_action = '';
+			// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Reason: We are not processing form information.
+			if ( isset( $_POST['action'] ) && \is_string( $_POST['action'] ) ) {
+				// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Reason: We are not processing form information.
+				$post_action = \sanitize_text_field( \wp_unslash( $_POST['action'] ) );
+			}
+			$option_page = '';
+			// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Reason: We are not processing form information.
+			if ( isset( $_POST['option_page'] ) && \is_string( $_POST['option_page'] ) ) {
+				// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Reason: We are not processing form information.
+				$option_page = \sanitize_text_field( \wp_unslash( $_POST['option_page'] ) );
```

Comment:
Do we really need sanitization since we're just comparing with literals and not storing/putting out anywhere?